### PR TITLE
Render video description

### LINKED ISSUE #27: Render video description too
Can probably show it in a collapsed div, just like youtube does beneath the video. It probably need to be eval'd as HTML

### PR DESCRIPTION
**Pull request by BeigeBadger**
Fixes #27 Render video description.

### CODE REVIEW

File: js/everything.js [hunk 1]
```diff
@@ -207,9 +208,27 @@ function makeListenURL(videoID) {
...
 
+function getVideoDescription(videoID) {
+    $.getJSON("https://www.googleapis.com/youtube/v3/videos", {
+        key: youTubeDataApiKey,
+        part: "snippet",
+        fields: "items/snippet/description",
+        id: videoID
+    }, function(data) {
+        if (data.items.length === 0) {
+            showErrorMessage("Video not found");
```
Comment (shakeelmohamed) [inline js/everything.js]: Should say `"Video description not found"`

**Comment by shakeelmohamed** [comment]
Awesome, thank you so much for starting this! I will create an API key :smile_cat: 

Can we clean this up a bit?

0. Can we parse the JSON response when there's an error and just display the human readable part?
0. The error message gets cleared but the error box is still visible (see gif below). The gif shows your changes as is against the `file://` protocol locally, which is probably the cause of the error.

test

**Comment by BeigeBadger** [comment]
I have addressed the issues above. :+1:

File: js/everything.js [hunk 2]
```diff
@@ -147,7 +148,7 @@ var VOLUME_LOCKED = false;
 function onPlayerReady(event) {
     // Only play the video if it's actually there
     if (getCurrentVideoID()) {
-        $("#zen-video-error").text("");
+        hideErrorMessage();
```
Comment (shakeelmohamed) [inline js/everything.js]: The error message will flash then disappear wen failing to get the video description here.
Maybe we should set a global var on line 233 for the description error, then only do `$("#zen-video-error").text("")` if that var is unset. My goal here is to persist errors.

What do you think?

File: index.html [hunk 3]
```diff
@@ -53,6 +53,7 @@
                 <br>
             </div>
             <h3><a id="zen-video-title"></a></h3>
+            <div id="zen-video-description"></div>
```
Comment (shakeelmohamed) [inline index.html]: Let's move this down between #playerControls and #demo, and probably want a `<br>` above & below, including the one that's already there

**Comment by shakeelmohamed** [comment]
Can you also add the following to the stylesheet?


```css
#zen-video-description {
    white-space: pre;
}
```

Before:
capturfiles_11


After:

capturfiles_10

File: index.html (hunk 3 above)
Comment (shakeelmohamed) [inline index.html]: How do you feel about adding a button to toggle the description like we have for player? (ie: Show/Hide Description)

**Comment by shakeelmohamed** [comment]
Hi @BeigeBadger, are you still interested in finishing this one?

**Comment by BeigeBadger** [comment]
Hi @shakeelmohamed, yes I am. Life has been a bit crazy lately (building a house at the moment) so I will try to finish it off this week.

**Comment by shakeelmohamed** [comment]
No problem at all @BeigeBadger, just checking in :rocket:

File: js/everything.js (hunk 2 above)
Comment (BeigeBadger) [inline js/everything.js]: Still TODO

**Comment by BeigeBadger** [comment]
I believe the last thing left TODO is https://github.com/zen-audio-player/zen-audio-player.github.io/pull/52#discussion_r43569871

File: index.html [hunk 4]
```diff
@@ -71,6 +71,9 @@
                 <i class="fa fa-volume-up fa-lg"></i>
             </div>
             <br>
+            <br>
```
Comment (shakeelmohamed) [inline index.html]: Remove this one

**Comment by shakeelmohamed** [comment]
Awesome work @BeigeBadger! The only things I see remaining:

0. A show/hide description button - https://github.com/zen-audio-player/zen-audio-player.github.io/pull/52#discussion_r43569955
0. Error persisting as you mentioned
0. What's a sane workaround for running in the `file://` protocol? Maybe we add a simple check like `window.location.protocol === "file:"` to skip the description request, then log to the console

I'm happy to take on these, let me know either way :+1:

File: index.html [hunk 5]
```diff
@@ -71,6 +71,14 @@
...
+            <div id="zen-video-description"></div>
+            <br>
+            <button class="btn" id="toggleDescription">
```
Comment (shakeelmohamed) [inline index.html]: Let's move this inline with `#togglePlayer`

File: index.html [hunk 6]
```diff
@@ -71,6 +71,14 @@
...
+                Show Description
+            </button>
+            <br/>
```
Comment (shakeelmohamed) [inline index.html]: After we move `#toggleDescription`, I think there should only be one `<br>` on either side of `#zen-video-description`

**Comment by shakeelmohamed** [comment]
Looks great @BeigeBadger, just a couple nits for the HTML structure! Travis won't correctly run the tests until you merge master down due to b26fc2e5bbe0f20f0077b8c56d54fbdf96a5071c

**Comment by BeigeBadger** [comment]
No problem. All done!

**Comment by shakeelmohamed** [comment]
Perfect, thank you!

**Comment by BeigeBadger** [comment]
Haha awesome gif, I'll have to post it in our work slack channel.:shipit:

**Pull request merged**